packages/ember-glimmer/tests/integration/helpers/unbound-test.js

```diff
@@ -0,0 +1,545 @@
+    this.registerHelper('-capitalize', (args) => args[0].toUpperCase());
+
+    let fooBarInstance;
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        fooBarInstance = this;
+      },
+      foo: 'bork'
+    });
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: FooBarComponent,
+      template: `{{-capitalize (unbound foo)}}`
+    });
+
+    this.render(`{{foo-bar}}`);
+
+    this.assertText('BORK');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('BORK');
+
+    this.runTask(() => {
+      set(fooBarInstance, 'foo', 'oof');
+      this.rerender();
+    });
+
+    this.assertText('BORK');
+
+    this.runTask(() => set(fooBarInstance, 'foo', 'blip'));
+
+    this.assertText('BORK');
+
+    this.runTask(() => {
+      set(fooBarInstance, 'foo', 'bork');
+      this.rerender();
+    });
+
+    this.assertText('BORK');
+  }
+
+  ['@htmlbars sexpr in helper format does not update on parent re-render']() {
+    let fooBarInstance;
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        fooBarInstance = this;
+      },
+      foo: 'bork'
+    });
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: FooBarComponent,
+      template: `{{-capitalize (unbound (-doublize foo))}}`
+    });
+
+    this.registerHelper('-capitalize', (params) => params[0].toUpperCase());
+
+    this.registerHelper('-doublize', (params) => `${params[0]} ${params[0]}`);
+
+    this.render(`{{foo-bar}}`);
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => {
+      set(fooBarInstance, 'foo', 'oof');
+      this.rerender();
+    });
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => set(fooBarInstance, 'foo', 'blip'));
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => {
+      set(fooBarInstance, 'foo', 'bork');
+      this.rerender();
+    });
+
+    this.assertText('BORK BORK');
+  }
+
+  ['@htmlbars should be able to render an unbound helper invocation']() {
+    this.registerHelper('-repeat', ([value], { count }) => {
+      let a = [];
+      while (a.length < count) {
+        a.push(value);
+      }
+      return a.join('');
+    });
+
+    this.render(`{{unbound (-repeat foo count=bar)}} {{-repeat foo count=bar}} {{unbound (-repeat foo count=2)}} {{-repeat foo count=4}}`, {
+      foo: 'X',
+      numRepeatsBinding: 'bar',
+      bar: 5
+    });
+
+    this.assertText('XXXXX XXXXX XX XXXX');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('XXXXX XXXXX XX XXXX');
+
+    this.runTask(() => set(this.context, 'bar', 1));
+
+    this.assertText('XXXXX X XX XXXX');
+
+    this.runTask(() => set(this.context, 'bar', 5));
+
+    this.assertText('XXXXX XXXXX XX XXXX');
+  }
+
+  ['@htmlbars should be able to render an bound helper invocation mixed with static values']() {
```

Comment:
Port of https://github.com/emberjs/ember.js/blob/bf067f7370c3c3381352d68ef5dfe55d1178c6f7/packages/ember-htmlbars/tests/helpers/unbound_test.js#L340-L360